src/brainfuck/Interpreter.java

```diff
@@ -57,8 +64,11 @@ public void run(Machine machine) throws IOException {
 			if (logger != null) {
 				logger.logStep(i, instructions.get(i), machine);
 			}
-			if (machine.isReversed()) i--;
-			else i++;
+			if (machine.isJumping()){
+				i = jumptable.getJump(instructions.get(i),i).intValue();
```

Comment:
Is the `intValue()` really required? Unboxing should handle that.
Also the instruction should be stored in a variable rather than repeating `instructions.get(i)`.
The instruction is not even needed if there is only one Map and the check for a ConditionalJump is performed here (which should not be required itself).